### PR TITLE
Fix type links and names

### PR DESCRIPTION
**Pull request by luksa**
Addresses #450 and #451

### CODE REVIEW

**Comment by n3wscott** [comment]
This reads really nicely. My only comment is changing the logical section name to the variable name, example: _Input Parameters Object_ --> _InputParametersSchema Object_ reads more like code rather than spec. But I am not opposed to it, maybe others can comment on this point. 

LGTM

**Comment by duglin** [comment]
hadn't really notice the squashing of those words - now that you point it out I agree - let's add spaces so its less codey.

**Comment by luksa** [comment]
I've added back the spaces. 

Note that I didn't want to change the type names in the OpenAPI schema definition itself, so I kept `ServiceInstanceSchema` singular instead of changing it to `ServiceInstanceSchemas` as I suggested in #450. The singular form is perfectly fine, but I would like to point out that there is a slight inconsistency between `Schemas` and `ServiceInstanceSchema/ServiceBindingSchema`. Do we want to fix that?

Also, there's a mismatch between the spec and the OpenAPI schema in the `InputParametersSchema` object. The type is called `SchemaParameters` in the OpenAPI schema. I haven't renamed it, since #392 changes the name to "JSON Schema Objects" (this name needs to be changed as I pointed out in that PR).

**Comment by n3wscott** [comment]
LGTM

**Comment by pmorie** [comment]
LGTM

File: spec.md [hunk 1]
```diff
@@ -353,24 +353,24 @@ It is therefore RECOMMENDED that implementations avoid such strings.
 
 | Response Field | Type | Description |
 | --- | --- | --- |
-| services* | array-of-service-objects | Schema of service objects defined below. MAY be empty. |
+| services* | [Service](#service-object)\[\] | Schema of service objects defined below. MAY be empty. |
```
Comment (duglin) [inline spec.md]: Can you make these "An array of ..." to match how we normal talk about arrays?

**Comment by n3wscott** [comment]
LGTM

**Comment by duglin** [comment]
LGTM

**Comment by pmorie** [comment]
LGTM

**Comment by fmui** [comment]
LGTM

**Pull request merged**